book/visual-effects.md

```diff
@@ -1493,10 +1493,6 @@ fits perfectly. In code, destination-in looks like this:
 class Pixel:
     def destination_in(self, source):
         self.a = self.a * source.a
-        if self.a == 0: return self
-        self.r = (self.r * self.a * source.a) / self.a
```

Comment:
This line simplifies to self.r = self.r * source.a, not a no-op...

Reply:
the link here seems to agree with what I said:

https://drafts.fxtf.org/compositing-1/#porterduffcompositingoperators_dstin

Reply:
Note that this only makes a difference if the alpha of the source is something other than 1 or 0, which we don't use in this chapter. But still.

Reply:
On line 1495 that should be new_a, not self.a, then it does cancel. Standard maybe different because premultiplied.